pandas/core/dtypes/cast.py

```diff
@@ -562,9 +562,16 @@ def maybe_promote(dtype: np.dtype, fill_value=np.nan):
         If fill_value is a non-scalar and dtype is not object.
     """
     orig = fill_value
+    orig_is_nat = False
     if checknull(fill_value):
         # https://github.com/pandas-dev/pandas/pull/39692#issuecomment-1441051740
         #  avoid cache misses with NaN/NaT values that are not singletons
+        if fill_value is not NA:
+            try:
+                orig_is_nat = np.isnat(fill_value)
+            except TypeError:
+                pass
```

Comment:
this seems fine, but is making me wonder if we should remove the caching on maybe_promote

Reply:
It appears this was done for performance reasons(?), but it certainly makes this a little more complex